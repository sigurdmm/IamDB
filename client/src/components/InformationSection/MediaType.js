import PropTypes from 'prop-types';
import React from 'react';

const MediaType = ({ type }) => <h3>Type of media: {type || 'unknown'}</h3>;

MediaType.propTypes = {
  type: PropTypes.string.isRequired,
};

export default MediaType;
