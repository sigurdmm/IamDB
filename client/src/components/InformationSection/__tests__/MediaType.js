import React from 'react';
import { render } from 'enzyme/build';
import MediaType from '../MediaType';

const mockedTypes = {
  existingType: 'Existing type',
  noType: {},
};

describe('<MediaType/>', () => {
  it('should render type when type is set', () => {
    const wrapper = render(<MediaType type={mockedTypes.existingType}/>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should render type when type is unknown', () => {
    const wrapper = render(<MediaType type={mockedTypes.noType}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
