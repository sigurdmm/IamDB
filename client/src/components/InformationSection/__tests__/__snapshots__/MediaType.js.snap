// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaType/> should render type when type is set 1`] = `
<h3>
  Type of media: Existing type
</h3>
`;
